### PR TITLE
feat(dockerfile): update alpine and cleanup the dockerfile

### PR DESCRIPTION
**Pull request by ellerbrock**
Hello @konstruktoid,

i updated the dockerfile and made few optimizations:

- update alpine to version 3.7
- combined the RUN commands to minimize the number of layer for the image
- add the copy command before the run command and get rid of the mkdir instead
- combined the directories query used in the rm command 
- removed the unused fake healthcheck

Have a nice weekend!

welcome

Cheers Maik

### CODE REVIEW

**Comment by konstruktoid** [comment]
Hi and thanks for the PR @ellerbrock, I'm having issues building and testing due to Alpine right now but I'll comment and hopefully the build will work soon(tm).

```fetch https://alpine.global.ssl.fastly.net/alpine/v3.7/main/x86_64/APKINDEX.tar.gz
fetch https://alpine.global.ssl.fastly.net/alpine/v3.7/community/x86_64/APKINDEX.tar.gz
WARNING: Ignoring https://alpine.global.ssl.fastly.net/alpine/v3.7/main/x86_64/APKINDEX.tar.gz: temporary error (try again later)
WARNING: Ignoring https://alpine.global.ssl.fastly.net/alpine/v3.7/community/x86_64/APKINDEX.tar.gz: temporary error (try again later)
```

File: Dockerfile [hunk 1]
```diff
@@ -1,27 +1,23 @@
...
-  mkdir /usr/local/bin/tests
-
-COPY ./*.sh /usr/local/bin/
-
-COPY ./tests/*.sh /usr/local/bin/tests/
+  rm -rf /usr/bin/docker?*
 
 WORKDIR /usr/local/bin
 
-HEALTHCHECK CMD exit 0
```
Comment (konstruktoid) [inline Dockerfile]: This is https://github.com/docker/docker-bench-security/pull/220, I'll test if it's still required.

File: distros/Dockerfile.alpine [hunk 2]
```diff
@@ -1,24 +1,23 @@
...
   apk add --no-cache \
     docker \
     dumb-init && \
-  rm -rf /usr/bin/docker-* /usr/bin/dockerd && \
-  mkdir /usr/local/bin/tests
-
-COPY ./*.sh /usr/local/bin/
-
-COPY ./tests/*.sh /usr/local/bin/tests/
+  rm -rf /usr/bin/docker?*
```
Comment (konstruktoid) [inline distros/Dockerfile.alpine]: Not really sure about this one.
```$ ls -1 /usr/bin/docker?*
/usr/bin/docker-containerd
/usr/bin/docker-containerd-ctr
/usr/bin/docker-containerd-shim
/usr/bin/dockerd
/usr/bin/docker-init
/usr/bin/docker-proxy
/usr/bin/docker-runc
$ sudo rm -rf /usr/bin/docker?*
$ ls -1 /usr/bin/docker?*
ls: cannot access '/usr/bin/docker?*': No such file or directory
$
```

File: Dockerfile [hunk 3]
```diff
@@ -1,27 +1,23 @@
...
   apk add --no-cache \
     docker \
     dumb-init && \
-  rm -rf /usr/bin/docker-* /usr/bin/dockerd && \
-  mkdir /usr/local/bin/tests
-
-COPY ./*.sh /usr/local/bin/
-
-COPY ./tests/*.sh /usr/local/bin/tests/
+  rm -rf /usr/bin/docker?*
```
Comment (konstruktoid) [inline Dockerfile]: Not really sure about this one.
```
$ ls -1 /usr/bin/docker?*
/usr/bin/docker-containerd
/usr/bin/docker-containerd-ctr
/usr/bin/docker-containerd-shim
/usr/bin/dockerd
/usr/bin/docker-init
/usr/bin/docker-proxy
/usr/bin/docker-runc
$ sudo rm -rf /usr/bin/docker?*
$ ls -1 /usr/bin/docker?*
ls: cannot access '/usr/bin/docker?*': No such file or directory
$
```

Comment (ellerbrock) [reply Dockerfile]: hey @konstruktoid,

yeah thats purpose :)
so with `rm -rf /usr/bin/docker?*` you deleted all that stuff.
when doing a `ls -l /usr/bin/docker*` and there should be only docker be left.

cheers maik

Comment (konstruktoid) [reply Dockerfile]: ah yes, my bad.

File: Dockerfile (hunk 1 above)
Comment (konstruktoid) [reply Dockerfile]:
```
[WARN] 4.6  - Ensure HEALTHCHECK instructions have been added to the container image
[WARN]      * No Healthcheck found: [docker-bench-security-ellerbrock:latest]
[WARN]      * No Healthcheck found: [alpine:3.7]
[WARN]      * No Healthcheck found: [alpine:3.6]
```

**Comment by konstruktoid** [comment]
Please include the fake health check (https://github.com/docker/docker-bench-security/pull/272/files#diff-3254677a7917c6c01f55212f86c57fbfL24) and I'm ready to merge.

**Comment by ellerbrock** [comment]
done

**Comment by konstruktoid** [comment]
Thanks!

**Pull request merged**